File: freight-triage-polyglot-ledger-recovery/environment/native/src/format/sign_prefix.cpp
#include "freight/formats.h"

#include <algorithm>
#include <cstdio>

namespace freight {

// Formatter: sign prefix.
std::string fmt_sign_prefix(long long value) {
  if (value == 0) {
    return "0";
  }
  if (value > 0) {
    return "+" + std::to_string(value);
  }
  return std::to_string(value);
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/format/SignPrefix.java
package com.freight.format;

/** Formatter: sign prefix. */
public final class SignPrefix implements Formatter {

    private static final String BASE36_DIGITS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    @Override
    public String name() {
        return "sign_prefix";
    }

    @Override
    public String apply(long value) {
        if (value == 0L) {
            return "0";
        }
        if (value > 0L) {
            return "+" + value;
        }
        return Long.toString(value);
    }
}
